### PR TITLE
added MD22 dataset node

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
This PR adds a Node for downloading and processing the MD22 datasets as well as an integration test.

### CODE REVIEW

**Comment by PythonFZ** [comment]
Now that we have the ability to self-host datasets through minio, wouldn't it be better to have the data somewhere in e.g. `ipsprojects/datasets` and load from there compared to introducing a Node for potentially many new datasets?

In code:

```python
data = zntrack.from_rev("MD22", remote=...)
```

vs

````python
ips.datasets.MD22()
````

**Comment by M-R-Schaefer** [comment]
idk what the legal aspects are of hosting other people's datasets.

File: ipsuite/datasets/md22.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            modified_line = line.replace(target_string, replacement_string)
+            output_file.write(modified_line)
+    return new_file_path
+
+
+def download_data(url: str, data_path: Path):
+    url_path = Path(urllib.parse.urlparse(url).path)
+    zip_path = data_path / url_path.stem
+    file_path = zip_path.with_suffix(".xyz")
+    urllib.request.urlretrieve(url, zip_path)
```
Comment (PythonFZ) [inline ipsuite/datasets/md22.py:28]: Do we want to keep the raw data? Maybe we could use `with tempfile` or `BytesIO`  to process the dataset in memory

**Review by PythonFZ — COMMENTED** [review]
small comment / question

File: ipsuite/datasets/md22.py (hunk 1 above)
Comment (M-R-Schaefer) [reply ipsuite/datasets/md22.py:28]: that's a good idea

**Review by PythonFZ — APPROVED** [review]

**Pull request merged**